src/js/actions/export.js

```diff
@@ -358,23 +359,41 @@ define(function (require, exports) {
     exportAllAssets.transfers = [updateLayerExportAsset];
 
     /**
-     * before start up, initialize the export service
+     * Before start up, Ensure that generator is enabled, and then initialize the export service
      * @return {Promise}
      */
     var beforeStartup = function () {
-        _exportService = new ExportService();
-        return _exportService.init()
+        return descriptor.playObject(generatorLib.getGeneratorStatus())
```

Comment:
I feel rather strongly that this should go in `afterStartup`, not `beforeStartup`. The UI and store should be robust to Generator not being enabled for some period of time, which means we shouldn't block rendering the UI while waiting for Generator to start. As a rule, if you don't want to block showing the UI, do it `afterStartup`.